comp/dogstatsd/replay/writer.go

```diff
@@ -202,32 +202,37 @@ func (tc *TrafficCaptureWriter) Capture(target io.WriteCloser, d time.Duration,
 		tc.oobPacketPoolManager.SetPassthru(false)
 	}
 
-	done := make(chan struct{})
-	defer close(done)
-	go func() {
-		log.Debugf("Capture will be stopped after %v", d)
-
-		select {
-		case <-time.After(d):
-			tc.StopCapture()
-		case <-done:
+	delay := time.After(d)
+Loop:
+	for {
+		if !tc.accepting {
+			// Drain the channel of any msgs that were enqueued in between the last message getting drained
+			// and the timer going off.
```

Comment:
I may be missing something or misreading this, but this doesn't match the code. We don't get into this branch after timer fired for the ongoing capture nor when a subsequent is started.

The only way to get here as far as I can see is to call `StopCapture` concurrently with the capture, which in this branch only happens on error and the agent shutdown.